ego/cli/sign.go

```diff
@@ -152,12 +152,13 @@ func (c *Cli) readConfigJSONtoStruct(path string) (*config.Config, error) {
 	conf.Key = filepath.Join(dir, conf.Key)
 
 	if err := conf.PopulateContent(c.fs); err != nil {
-		return nil, err
+		// Wrap error since certain callers may ignore os.IsNotExist to create a new manifest
+		return nil, fmt.Errorf("failed to populate embedded file content: %v", err)
```

Comment:
should we go the other way round and return a custom error if ReadFile(path) fails with NotExist?

Reply:
I have a slight preference for wrapping PopulateContent's error since it's imo not super well positioned in this function and provides some context that we fail inside the file embedding.

Though the error does usually contain the offending filename anyway so it's not a huge pro argument imo.

What would be the advantage of going the other way around? More specific error handling rather than the more generic approach we currently have?

Reply:
Yes, someone may introduce a similar bug in the future by adding something that could return NotExist. But I also get your point. I leave it up to you.

Reply:
I'll do both I guess. My current change for clarity that something is wrong with the files to be embedded, the other one for more specific error handling.